pkg/controller.v1alpha3/suggestion/suggestion_controller.go

```diff
@@ -134,7 +134,16 @@ func (r *ReconcileSuggestion) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 	instance := oldS.DeepCopy()
+	// If ResumePolicyType is LongRunning, suggestion status will never be succeeded.
 	if instance.IsCompleted() {
```

Comment:
Instance also is completed when Suggestion is failed: https://github.com/kubeflow/katib/blob/master/pkg/apis/controller/suggestions/v1alpha3/util.go#L67. And this can happen when `ValidateAlgorithmSettings` is failed. Do we want to delete deployment in that case?

Reply:
I'm not sure why we need to remain the failed deployment, may be for debug?

Reply:
Yes, e.g. to check logs from Suggestion deployment.